lib/ImageHelper.php

```diff
@@ -170,24 +170,10 @@ public static function img_to_jpg( $src, $bghex = '#FFFFFF', $force = false ) {
 	 */
 	protected static function add_actions() {
 		add_action('delete_attachment', function( $post_id ) {
-			$post = get_post($post_id);
-			$image_types = array('image/jpeg', 'image/png', 'image/gif', 'image/jpg');
-			if ( in_array($post->post_mime_type, $image_types) ) {
-				$attachment = new Image($post_id);
-				if ( $attachment->file_loc ) {
-					ImageHelper::delete_generated_files($attachment->file_loc);
-				}
-			}
+			ImageHelper::delete_generated_if_image($post_id);
 		} );
 		add_filter('wp_generate_attachment_metadata', function( $metadata, $post_id ) {
-			$post = get_post($post_id);
-			$image_types = array('image/jpeg', 'image/png', 'image/gif', 'image/jpg');
-			if ( in_array($post->post_mime_type, $image_types) ) {
-				$attachment = new Image($post_id);
-				if ( $attachment->file_loc ) {
-					ImageHelper::delete_generated_files($attachment->file_loc);
-				}
-			}
+			ImageHelper::delete_generated_if_image($post_id);
 			return $metadata;
 		} );
```

Comment:
When using `add_filter` with a function that uses more than one parameter, you need to state the amount of parameters that you want to pass to the function. In this case, it’s `2` parameter with the default priority of `10`.

```php
add_filter('wp_generate_attachment_metadata', function($metadata, $post_id) {
	// Your function content
}, 10, 2);
```